conanfile.py

```diff
@@ -10,9 +10,15 @@ class ProtobufConan(ConanFile):
     version = "2.6.1"
     url = "https://github.com/memsharded/conan-protobuf.git"
     settings = "os", "compiler", "build_type", "arch"
-    exports = "CMakeLists.txt", "*.cmake", "extract_includes.bat.in"
+    exports = "CMakeLists.txt", "*.cmake", "extract_includes.bat.in", "change_dylib_names.sh"
     options = {"static": [True, False]}
     default_options = "static=False"
+    requires = "zlib/1.2.8@lasote/stable"
+    generators = "cmake"
+    license = "https://github.com/google/protobuf/blob/v2.6.1/LICENSE"
+
+    def config(self):
```

Comment:
It is likely that the latest can already be achieved with the ``conan_info()`` method. Basically this method allows to redefine the options, settings and requirements that define the SHA signature. It is intended to allow narrowing package variability. Typical use case would be, that the input setting is "gcc, 4.8", but the package creator is absolutely sure that the package built with 4.8 will be also compatible, and can be linked when building with gcc 4.9 and viceversa. So they could do something like to always obtain the same SHA, for both versions:

```python
def conan_info(self):
    if self.info.settings.compiler.version == "4.8" or self.info.settings.compiler.version == "4.8":
        self.info.settings.compiler.version = "4.8-9"  # or any other syntax
```


By default, the options are stripped to use just the self package options, but the upstream options could be easily restored if they affect the package:

```python
def conan_info(self):
    self.info.options = self.info.full_options
    # Probably something like the following could be used to set just 1 option
    # self.info.options["Boost"].shared = self.info.full_options["Boost"].shared
```

However, I have checked that other packages as boost are using the same pattern, so I am not totally sure which would be the best, will have another look.